pydantic_ai_slim/pydantic_ai/profiles/_json_schema.py

```diff
@@ -76,7 +75,10 @@ def _handle(self, schema: JsonSchema) -> JsonSchema:
         nested_refs = 0
         if self.prefer_inlined_defs:
             while ref := schema.get('$ref'):
-                key = re.sub(r'^#/\$defs/', '', ref)
+                if ref.startswith('#/$defs/'):
+                    key = ref[8:]  # len('#/$defs/')
+                else:
+                    key = re.sub(r'^#/\$defs/', '', ref)
```

Comment:
I'm pretty sure this else statement is unnecessary; more generally imo this change is making things slightly less readable for presumably negligible performance consequences in real world usage (10x faster on something that takes 2ms over app lifetime doesn't really matter; maybe in practice it is heavier than this but ultimately what I care about is real world performance impact).